generators/app/templates/infrastructure/src/utils/apiRestDataSource.js

```diff
@@ -0,0 +1,28 @@
+const { correlationManager } = require("@totalsoft/correlation");
+const NoCacheRESTDataSource = require("./noCacheRESTDataSource");
+const { TenantId, UserId, UserPassport } = require("../constants/customHttpHeaders");
+
+class ApiRESTDataSource extends NoCacheRESTDataSource {
+
+  willSendRequest(request) {
+    <%_ if(withMultiTenancy){ _%>
+    request.headers.set(TenantId, this.context.state?.jwtdata?.tid ?? "");
+    <%_}_%>
+    request.headers.set(UserPassport, this.context.state?.jwtdata ? JSON.stringify(this.context.state.jwtdata) : "");
+    request.headers.set(UserId, this.context.state?.jwtdata?.sub ?? "");
```

Comment:
I would write this sequence of code like this:
```javascript
const { jwtdata } = this.context.state ?? {}
 <%_ if(withMultiTenancy){ _%>
request.headers.set(TenantId, jwtdata?.tid ?? "");
 <%_}_%>
request.headers.set(UserPassport, JSON.stringify(jwtdata ?? ""));
request.headers.set(UserId, jwtdata?.sub ?? "");
```
Can a value from the header be set to `undefined` or the `""` is mandatory? If the empty string is not mandatory, I guess you can remove all the `?? ""` occurrences.

Reply:
JSON.stringify("") results in '""' string, not ""

Reply:
JSON.stringify("") results in '""' string, not empty string